src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClientTest.java

```diff
@@ -163,97 +175,63 @@ void should_throw_exception_when_token_decoding_fails() {
     void should_create_token_when_cache_is_expired() throws InterruptedException {
 
         IdamCachedClient idamCachedClientQuickExpiry = new IdamCachedClient(
-            idamClient,
+            idamApi,
             users,
-            new AccessTokenCacheExpiry(28798)
+            new IdamCacheExpiry(28798)
         );
 
         String jurisdiction = "probate";
 
         given(users.getUser(jurisdiction)).willReturn(new Credential(USERNAME, PASSWORD));
-        given(idamClient.authenticateUser(USERNAME, PASSWORD)).willReturn(JWT, JWT2);
-        String token1 =
-            idamCachedClientQuickExpiry.getAccessToken(jurisdiction);
+        given(idamApi.authenticateUser(USERNAME, PASSWORD)).willReturn(JWT, JWT2);
+        given(idamApi.getUserDetails(JWT)).willReturn(USER_DETAILS);
+        UserDetails expectedUserDetails2 =  new UserDetails("12","q@a.com","","",null);
+        given(idamApi.getUserDetails(JWT2)).willReturn(expectedUserDetails2);
+
+        CachedIdamCredential cachedIdamCredential1 =
+            idamCachedClientQuickExpiry.getIdamCredentials(jurisdiction);
 
         TimeUnit.SECONDS.sleep(3);
```

Comment:
Can we make it expire even quicker?

Reply:
expiry  is 2 second. I want to be sure expired as it is configured.

Reply:
worth adding a comment here why this sleep is added

Reply:
comment added